jupyterlab_gcedetails/src/components/hardware_scaling_status.spec.tsx

```diff
@@ -0,0 +1,218 @@
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    expect(hardwareScalingStatus.state('status')).toEqual(Status.Error);
+    expect(hardwareScalingStatus).toMatchSnapshot('Auth Error');
+  });
+
+  it('Renders with stop error', async () => {
+    mockAuthTokenRetrieval.mockResolvedValue(mockToken);
+    mockStopInstance.mockImplementation(() => {
+      throw new Error();
+    });
+    const detailsResponse = await JSON.parse(DETAILS_RESPONSE);
+    const hardwareConfig = detailsToHardwareConfiguration(detailsResponse);
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    await mockAuthTokenRetrieval();
+    expect(hardwareScalingStatus.state('status')).toEqual(Status.Error);
+    expect(hardwareScalingStatus).toMatchSnapshot('Stop instance error');
+  });
+
+  it('Renders with operation error', async () => {
+    const mockGetUtilizationData = jest.fn();
+    const mockStopInstance = jest.fn();
+    mockSetMachineType.mockImplementation(() => {
+      throw new Error();
+    });
+    mockSetAccelerator.mockImplementation(() => {
+      throw new Error();
+    });
+    mockStartInstance.mockResolvedValue(mockInstance);
+    const detailsResponse = await JSON.parse(DETAILS_RESPONSE);
+    const resolveValue = Promise.resolve({ ...detailsResponse });
+    mockGetUtilizationData.mockReturnValue(resolveValue);
+    const hardwareConfig = detailsToHardwareConfiguration(detailsResponse);
+    const hardwareScalingStatus = shallow(
+      <HardwareScalingStatus
+        detailsServer={mockServerWrapper}
+        notebookService={mockNotebookService}
+        onCompletion={mockOnComplete}
+        onDialogClose={mockOnDialogClose}
+        hardwareConfiguration={hardwareConfig}
+        authTokenRetrieval={mockAuthTokenRetrieval}
+      />
+    );
+    await mockAuthTokenRetrieval();
+    await mockStopInstance();
+    await mockStartInstance();
+    await mockGetUtilizationData();
+    setTimeout(() => {
+      expect(hardwareScalingStatus.state('status')).toEqual(Status.Error);
+      expect(hardwareScalingStatus).toMatchSnapshot('Operation error');
+      hardwareScalingStatus
+        .find(Button)
+        .first()
+        .simulate('click');
+      expect(mockOnDialogClose).toBeCalled();
+    }, 1000);
+  });
+
+  it('Renders with restart error', async () => {
```

Comment:
So I tried to go for a testing approach which doesn't cover every single error situation, but it covers all the ones where the UI would be different. Let me know if it should be exhaustive though

Reply:
I think this is sufficient as each error is handled similarly regardless of where it occurs.